cmd/argo/commands/server.go

```diff
@@ -191,18 +190,8 @@ See %s`, help.ArgoServer),
 		},
 	}
 
-	defaultBaseHRef := os.Getenv("BASE_HREF")
-	if defaultBaseHRef == "" {
-		defaultBaseHRef = "/"
-	}
-
-	defaultAllowedLinkProtocol := []string{"http", "https"}
-	if protocol := os.Getenv("ALLOWED_LINK_PROTOCOL"); protocol != "" {
-		defaultAllowedLinkProtocol = strings.Split(protocol, ",")
-	}
-
 	command.Flags().IntVarP(&port, "port", "p", 2746, "Port to listen on")
-	command.Flags().StringVar(&baseHRef, "basehref", defaultBaseHRef, "Value for base href in index.html. Used if the server is running behind reverse proxy under subpath different from /. Defaults to the environment variable BASE_HREF.")
```

Comment:
@agilgur5 This is a breaking change. `Need to be marked` as breaking change.
where does the code refer to `ARGO_BASE_HREF` from Viber to make it base href? 
Am I missing any change?

Reply:
It is marked as a breaking change -- the PR has a `!`, the description says so, and the docs for breaking changes was updated in this PR.

Regarding Viper, that too is described in the PR and was described in my [previous comment](https://github.com/argoproj/argo-workflows/pull/12653#discussion_r1487240445) as well

Reply:
Thanks @agilgur5  sorry I missed the description and `!`. 


As per code `ARGO_BASE_HREF` will not considered during `argo server` starting. 
This code was used to get `BASE_HREF` from `env` and updated during server start. It could be a bug in existing code.  This [code](https://github.com/argoproj/argo-workflows/blob/986b069f34ed5336b6f754cfc6709c30b302c768/cmd/argo/commands/client/conn.go#L57) was never called during the server start.
```go
	defaultBaseHRef := os.Getenv("BASE_HREF")
	if defaultBaseHRef == "" {
		defaultBaseHRef = "/"
	}
```
I tried in my local base href didn't update. Please try on your env set `ARGO_BASE_HREF` and start the `argo-server` to check `baseHRef` updating
```log
(base) ➜  argo-workflows git:(main) ✗ export ARGO_BASE_HREF="/test"
(base) ➜  argo-workflows git:(main) ✗ dist/argo server             
INFO[2024-02-18T20:51:52.092Z] not enabling pprof debug endpoints           
INFO[2024-02-18T20:51:52.096Z]                                               authModes="[client]" baseHRef=/ managedNamespace= namespace=argo secure=true ssoNamespace=argo
INFO[2024-02-18T20:51:52.096Z] Generating Self Signed TLS Certificates for Secure Mode 
INFO[2024-02-18T20:51:52.096Z] SSO disabled                                 
INFO[2024-02-18T20:51:52.109Z] Starting Argo Server                          instanceID= version=untagged
INFO[2024-02-18T20:51:52.109Z] Creating event controller                     asyncDispatch=false operationQueueSize=16 workerCount=4
INFO[2024-02-18T20:51:52.111Z] GRPC Server Max Message Size, MaxGRPCMessageSize, is set  GRPC_MESSAGE_SIZE=104857600
INFO[2024-02-18T20:51:52.111Z] Argo Server started successfully on https://localhost:2746  url="https://localhost:2746"
```

Reply:
> This code was used to get `BASE_HREF` from `env` and updated during server start.

Yes, I'm aware, I intentionally removed it since Viper does that automatically.

> This [code](https://github.com/argoproj/argo-workflows/blob/986b069f34ed5336b6f754cfc6709c30b302c768/cmd/argo/commands/client/conn.go#L57) was never called during the server start.

That's client code, not Server code. That's used for the CLI commands that connect to the Server

> I tried in my local base href didn't update. Please try on your env set `ARGO_BASE_HREF` and start the `argo-server` to check `baseHRef` updating

👀  let me double-check that. I'm pretty sure I've used this a few times before

> It could be a bug in existing code.

Maybe, I'll check

Reply:
Ok apparently, there's a bit of a typo in the old code... it's [`--basehref`](https://github.com/argoproj/argo-workflows/blob/986b069f34ed5336b6f754cfc6709c30b302c768/cmd/argo/commands/server.go#L205) in the Server, not `--base-href`, so the env var is `ARGO_BASEHREF`.

This is pretty confusing given that the client uses [`--argo-base-href`](https://github.com/argoproj/argo-workflows/blob/986b069f34ed5336b6f754cfc6709c30b302c768/cmd/argo/commands/client/conn.go#L48).

I would like to make them both use dash-case/kebab-case, which would convert to `ARGO_BASE_HREF`, but that would be another breaking change 😕

Reply:
Ok I changed it to `--base-href` for consistency and added another breaking change note